--- a/net/freeradius3/Config.in
+++ b/net/freeradius3/Config.in
@@ -3,7 +3,6 @@
 choice
 	prompt "SSL library"
 	default FREERADIUS3_OPENSSL
-	depends on PACKAGE_freeradius3-common
 
 config FREERADIUS3_NOSSL
 	bool "No SSL support"
--- a/net/freeradius3/Makefile
+++ b/net/freeradius3/Makefile
@@ -63,7 +63,7 @@
 define Package/freeradius3-common
   $(call Package/freeradius3/Default)
   TITLE:=common files
-  DEPENDS:=+USE_GLIBC:libcrypt-compat +USE_GLIBC:libpthread +USE_GLIBC:libbsd +FREERADIUS3_OPENSSL:libopenssl +FREERADIUS3_OPENSSL:libopenssl-legacy +libcap +libpcap +libncurses +libreadline +libtalloc +libatomic
+  DEPENDS:=+USE_GLIBC:libcrypt-compat +USE_GLIBC:libpthread +USE_GLIBC:libbsd +libopenssl +libopenssl-legacy +libcap +libpcap +libncurses +libreadline +libtalloc +libatomic
 endef
 
 define Package/freeradius3-default
